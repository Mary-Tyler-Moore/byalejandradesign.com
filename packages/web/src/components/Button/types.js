/** @flow */
import * as React from 'react';

export type Props = {
  children?: React.Node,
  type: 'submit' | 'reset' | 'button',
  className?: string,
  name?: string,
  onClick?: null | ((e: SyntheticEvent<HTMLButtonElement>) => any),
  margin?: boolean,
  fullWidth?: boolean,
};
