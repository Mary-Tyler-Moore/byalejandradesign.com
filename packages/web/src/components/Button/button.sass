@import ~@njmyers/sass/material/button
@import ../../sass/global

.defaultButton
  @extend %button-yellow

  width: 100%

  &:hover
    background-color: lighten($button-yellow, 6)

.greyButton
  @extend %button-grey

  width: 100%

.clearButton
  @extend %button-transparent

  border: 2px solid $white
  opacity: 0.9
  color: $white
  width: 100%

  &:hover
    background: transparentize($white, 0.75)
