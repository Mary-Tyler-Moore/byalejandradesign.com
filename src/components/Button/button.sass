@import ../Global/extras

.defaultButton
  @extend %button-styled
