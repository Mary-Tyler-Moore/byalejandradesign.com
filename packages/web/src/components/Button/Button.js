import * as React from 'react';
import './button.sass';

class Button extends React.Component {
  static defaultProps = {
    className: 'button-yellow',
    type: 'button',
  };

  render() {
    return (
      <button
        onClick={this.props.onClick}
        name={this.props.name}
        className={this.props.className}
        style={{
          width: this.props.fullWidth ? '100%' : 'inherit',
          margin: this.props.margin ? '1rem 0' : '0',
        }}
        type={this.props.type}
      >
        <span className="button_text-sourceSans">{this.props.children}</span>
      </button>
    );
  }
}

export default Button;
