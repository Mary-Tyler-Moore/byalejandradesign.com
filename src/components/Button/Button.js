import * as React from 'react';
import './button.sass';

class Button extends React.Component {
  static defaultProps = {
    className: 'defaultButton',
  };

  render() {
    return (
      <button
        onClick={this.props.onClick}
        name={this.props.name}
        className={this.props.className}
      >
        <span className="defaultButton_text">{this.props.children}</span>
      </button>
    );
  }
}

export default Button;
