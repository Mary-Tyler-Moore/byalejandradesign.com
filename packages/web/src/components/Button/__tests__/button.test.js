import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Button from '../Button';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<Button />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('it loads', () => {
    const render = TestRenderer.create(
      <Button margin fullWidth className="class" name="button" type="submit">
        click me
      </Button>
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
