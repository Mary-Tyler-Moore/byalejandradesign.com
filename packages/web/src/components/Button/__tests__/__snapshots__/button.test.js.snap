// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `
<button
  className="button-yellow"
  style={
    Object {
      "margin": "0",
      "width": "inherit",
    }
  }
  type="button"
>
  <span
    className="button_text-sourceSans"
  />
</button>
`;
